wrappers/src/fdw/logflare_fdw/logflare_fdw.rs

```
@@ -0,0 +1,317 @@
+use pgrx::{
+    pg_sys,
+    prelude::{AnyNumeric, Date, PgSqlErrorCode, Timestamp},
+};
+use reqwest::{
+    self,
+    header::{HeaderMap, HeaderName, HeaderValue},
+    StatusCode, Url,
+};
+use reqwest_middleware::{ClientBuilder, ClientWithMiddleware};
+use reqwest_retry::{policies::ExponentialBackoff, RetryTransientMiddleware};
+use serde_json::value::Value as JsonValue;
+use std::collections::HashMap;
+use std::str::FromStr;
+
+use supabase_wrappers::prelude::*;
+
+fn create_client(api_key: &str) -> ClientWithMiddleware {
+    let mut headers = HeaderMap::new();
+    let header_name = HeaderName::from_static("x-api-key");
+    let mut auth_value = HeaderValue::from_str(api_key).unwrap();
+    auth_value.set_sensitive(true);
+    headers.insert(header_name, auth_value);
+    let client = reqwest::Client::builder()
+        .default_headers(headers)
+        .build()
+        .unwrap();
+    let retry_policy = ExponentialBackoff::builder().build_with_max_retries(3);
+    ClientBuilder::new(client)
+        .with(RetryTransientMiddleware::new_with_policy(retry_policy))
+        .build()
+}
+
+macro_rules! report_request_error {
+    ($err:ident) => {{
+        report_error(
+            PgSqlErrorCode::ERRCODE_FDW_ERROR,
+            &format!("request failed: {}", $err),
+        );
+        return;
+    }};
+}
+
+fn extract_params(quals: &[Qual]) -> Option<Vec<Qual>> {
+    let params = quals
+        .iter()
+        .filter_map(|q| {
+            if q.field.starts_with("_param_") {
+                Some(q.clone())
+            } else {
+                None
+            }
```

does this silently ignore components of the `where` clause that don't start with `_param_` or are the other filters applied elsewhere?

---

no it won't, that is just to find `_param_` conditions in `where` clause and pass them to api call, the other conditions will still apply after data is retrieved from logflare.